src/libraries/System.IO.Pipes.AccessControl/ref/System.IO.Pipes.AccessControl.cs

```diff
@@ -71,4 +71,9 @@ public static class AnonymousPipeServerStreamAcl
     {
         public static System.IO.Pipes.AnonymousPipeServerStream Create(System.IO.Pipes.PipeDirection direction, System.IO.HandleInheritability inheritability, int bufferSize, System.IO.Pipes.PipeSecurity pipeSecurity) { throw null; }
     }
+
+    public static class NamedPipeServerStreamAcl
+    {
+        public static System.IO.Pipes.NamedPipeServerStream Create(string pipeName, System.IO.Pipes.PipeDirection direction, int maxNumberOfServerInstances, System.IO.Pipes.PipeTransmissionMode transmissionMode, System.IO.Pipes.PipeOptions options, int inBufferSize, int outBufferSize, System.IO.Pipes.PipeSecurity pipeSecurity, System.IO.HandleInheritability inheritability = System.IO.HandleInheritability.None, System.IO.Pipes.PipeAccessRights additionalAccessRights = default) { throw null; }
```

Comment:
Copying an interesting discussion from the old corefx PR.
tl;dr: This has been solved with a fix on arcade.

---

carlossanlop Nov 20, 2019:

```cs
            PipeAccessRights additionalAccessRights = default)
```

This argument is the reason why the unit tests failed for Linux when I submitted the first commit of the PR.

The autogenerated file `System.IO.Pipes.AccessControl.notsupported.cs` was not getting the `default` keyword added, so the build was failing in Unix architectures. The file looked like this:

```cs
            PipeAccessRights additionalAccessRights = )
```

This was happening because:

A) `PipeAccessRights` is an enum that does not set a value of zero (can't grant or deny 'nothing'). We have it documented in the source: [runtime/src/libraries/System.IO.Pipes/src/System/IO/Pipes/PipeAccessRights.cs#L10](https://github.com/dotnet/runtime/blob/be980b71efadc622b5720a36867696758e59e71c/src/libraries/System.IO.Pipes/src/System/IO/Pipes/PipeAccessRights.cs#L10)
B) Even though C# knows that the default value for an enum instance is zero, we did not know how to handle this case in arcade when autogenerating the `*.notsupported.cs` files, because we were making the assumption that all enums would define a value for zero (we were looking for a specific value).

So we fixed this in arcade with the help of Santi and Eric in PR https://github.com/dotnet/arcade/pull/4402 .

Eric found that our guidelines have specific recommendations for setting a value of zero to an enum:
https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/enum

>> X AVOID using flag enum values of zero unless the value represents "all flags are cleared" and is named appropriately, as prescribed by the next guideline.
>> ✓ DO name the zero value of flag enums None. For a flag enum, the value must always mean "all flags are cleared."

So the question is what to do about the `default` value we are assigning to `PipeAccessRights` in the public method's signature, which was accepted in the API proposal:

1. Should we add a value of `None` to the enum, considering it does not make sense to grant or deny 'nothing'?
2. Should we choose a different default value for that enum, like `ReadWrite`?  I don't like this option because maybe the user does not want to add any additional rights.
3. Should we set it to `(PipeAccessRights)0` instead of `default`?
4. Should we leave it as is, now that the arcade bug was fixed?

I had an internal email discussion with my team and got an answer from @bartonjs:

> _It sounds like the problem is just in one of the generators… specifying `default` should work fine to get the caller to pass 0 even on a non-conforming [Flags] enum (no named zero)._
> _If the question is “can/should we add the None = 0 enum field?” I’d be supportive of that, seems pretty uncontroversial._

I think `default` or `0` are ok values for the `additionalRights` parameter because, as the name states, they are extra and optional.

On the other hand, setting `PipeAccessRIghts` to `0` in a `PipeSecurity` instance doesn't make much sense, sounds like should not be allowed. I will write a unit test to see if we throw either when we create the `PipeSecurity` object or when we attempt to create a `NamedPipeServerStream` with a zero rights security object.